pyresample/gradient/__init__.py

```diff
@@ -137,12 +137,11 @@ def _get_src_poly(self, src_y_start, src_y_end, src_x_start, src_x_end):
         """Get bounding polygon for source chunk."""
         geo_def = self.source_geo_def[src_y_start:src_y_end,
                                       src_x_start:src_x_end]
-        try:
-            src_poly = get_polygon(self.prj, geo_def)
-        except AttributeError:
-            # Can't create polygons for SwathDefinition
-            src_poly = False
-
+        if isinstance(geo_def, SwathDefinition): 
+            return False 
+        
+        # NOTE: th code below could be used to return a polygon also for SwathDefinition object
+        src_poly = get_polygon(self.prj, geo_def)
```

Comment:
The failing GradientResampler tests were caused by this code line. Before  `get_polygon` was raising an `AttributeError` ... and so `False` was returned. Now `get_polygon` is able to return a polygon also for `SwathDefinition`, so I modified the code to keep the behaviour of returning False is `SwathDefinition`

Reply:
This sounds good to me. I think for Satpy use cases even though it is *possible* to return a polygon, most of the times it means computing a lot of data and that is overall slow. In the future, metadata from input data like g-ring/polygon information could be stored in the SwathDefinition (or something like it) and that would be the polygon returned here and would be fast. We're not there yet though.